src/jit/assertionprop.cpp

```diff
@@ -2215,9 +2215,14 @@ GenTreePtr Compiler::optVNConstantPropOnTree(BasicBlock* block, GenTreePtr stmt,
 #ifdef _TARGET_64BIT_           
             if (vnStore->IsVNHandle(vnCns))
```

Comment:
I added the asserts. The constants seen in gtFoldExprConst are under either GT_NOP of GT_EQ or GT_NE. The latter are tests we are generating for things like typeof(T) == typeof(int).